react/features/rtcstats/RTCStats.ts

```diff
@@ -185,6 +196,38 @@ class RTCStats {
         this.trace?.close();
     }
 
+    /**
+     * RTCStats client can notify the APP of any PeerConnection related event that occurs.
+     *
+     * @param {Object} event - Rtcstats event.
+     * @returns {void}
+     */
+    handleRtcstatsEvent(event: any) {
+        switch (event.type) {
+        case PC_CON_STATE_CHANGE: {
+            const { body: { isP2P, state } = { state: null,
+                isP2P: null } } = event;
+
+            this.connStateEvents.push(event.body);
+
+            // We only report PC related connection issues. If the rtcstats websocket is not connected at this point
+            // it usually means that none of our services can be reached i.e. there's problem with the internet
+            // connection and not necessarily with reaching the JVB (due to a firewall or other reasons).
+            if (state === PC_STATE_FAILED && this.trace.isConnected()) {
+                const connectionType = isP2P ? 'P2P' : 'JVB';
+                const wasConnected = this.connStateEvents.some((connectionEvent: { isP2P: any; state: string; }) =>
+                    (connectionEvent.isP2P === isP2P) && (connectionEvent.state === PC_STATE_CONNECTED));
+
+                logger.info(`${connectionType} PeerConnection failed, previously connected: ${wasConnected}`);
+
+                APP?.API?.notifyPeerConnectionFailure(isP2P, wasConnected);
```

Comment:
This won't cut it, you'd need to optional chain the function call too. Better to check if APP is undefined or not.

Reply:
I'm a bit confused, isn't the first part `APP?.API` the same as checking if APP is undefined or not?

Reply:
That works if there is a variable called APP which is undefined. But APP is not even declared.

Reply:
understood, fixed.